esiclient/tests/unit/v1/test_node_baremetal.py

```diff
@@ -0,0 +1,129 @@
+                                power_state="off", provision_state="active",
+                                maintenance=False)
+
+        self.node2 = munch.Munch(uuid="node_uuid_2", name="node2",
+                                 instance_uuid="instance_uuid_2",
+                                 power_state="off", provision_state="active",
+                                 maintenance=False)
+
+        self.node3 = munch.Munch(uuid="node_uuid_3", name="node3",
+                                 instance_uuid="instance_uuid_3",
+                                 power_state="off", provision_state="active",
+                                 maintenance=False)
+
+        self.node4 = munch.Munch(uuid="node_uuid_4", name="node4",
+                                 instance_uuid="instance_uuid_4",
+                                 power_state="off", provision_state="active",
+                                 maintenance=False)
+
+    @mock.patch('esiclient.v1.node_baremetal.openstack.connect')
+    @mock.patch('esiclient.v1.node_baremetal.openstack.config.loader.'
+                'OpenStackConfig.get_all_clouds')
+    def test_take_action_list(self, mock_config, mock_connect):
+
+        mock_cloud = mock.Mock()
+        mock_cloud.list_machines.side_effect = [[self.node], [self.node2],
+                                                [self.node3, self.node4]]
+        mock_connect.return_value = mock_cloud
+
+        mock_config.return_value = [self.cloud1, self.cloud2, self.cloud3]
+
+        arglist = []
+        verifylist = []
+
+        parsed_args = self.check_parser(self.cmd, arglist, verifylist)
+
+        results = self.cmd.take_action(parsed_args)
+        expected = (
+            ['Cloud', 'Region', 'UUID', 'Name', 'Instance UUID', 'Power State',
+             'Provisioning State', 'Maintenance'],
+            [
+                [self.cloud1._name, self.cloud1.config['region_name'],
+                 "node_uuid_1", "node1", "instance_uuid_1", "off", "active",
+                 False],
+                [self.cloud2._name, self.cloud2.config['region_name'],
+                 "node_uuid_2", "node2", "instance_uuid_2", "off", "active",
+                 False],
+                [self.cloud3._name, self.cloud3.config['region_name'],
+                 "node_uuid_3", "node3", "instance_uuid_3", "off", "active",
+                 False],
+                [self.cloud3._name, self.cloud3.config['region_name'],
+                 "node_uuid_4", "node4", "instance_uuid_4", "off", "active",
+                 False],
+            ]
+        )
+        self.assertEqual(expected, results)
+        mock_config.assert_called_once()
+        assert mock_cloud.list_machines.call_count == 3
+
+    @mock.patch('esiclient.v1.node_baremetal.openstack.connect')
+    @mock.patch('esiclient.v1.node_baremetal.openstack.config.loader.'
+                'OpenStackConfig.get_all_clouds')
+    def test_take_action_list(self, mock_config, mock_connect):
```

Comment:
This test has the same name as the other one; they should have different names.

Reply:
my bad, fixed